src/plugins/dataflow-tool/model/dataflow-program-model.ts

```diff
@@ -0,0 +1,134 @@
+import { types, Instance } from "mobx-state-tree";
+
+const ConnectionModel = types
+  .model("Connection", {
+    node: types.number,
+    output: types.maybe(types.string),
+    input: types.maybe(types.string),
+    data: types.map(types.string)
+  });
+  
+const SocketModel = types
+  .model("Socket", {
+    connections: types.map(ConnectionModel)
+  })
+  .preProcessSnapshot((snapshot: any) => {
+    // Connections are stored in rete as an array, but MST works better with maps
+    if (Array.isArray(snapshot.connections)) {
+      const connections: any = {};
+      snapshot.connections.forEach((connection: any) => {
+        connections[`${connection.node}-${connection.output}-${connection.input}`] = connection;
+      });
+      return { connections };
+    }
+    return snapshot;
```

Comment:
My comment about testing got dropped from the previous review for some reason. Model classes like these are good candidates for jest tests, which can test the transforming and non-transforming paths, for instance. I'd suggest adding some tests that at least verify that the pre- and post-processing are working as expected.